extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/quantiles/DoublesSketchAggregatorFactory.java

```diff
@@ -0,0 +1,267 @@
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.quantiles;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.IAE;
+import com.yahoo.sketches.Util;
+import com.yahoo.sketches.quantiles.DoublesSketch;
+import com.yahoo.sketches.quantiles.DoublesUnion;
+
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.aggregation.AggregatorFactory;
+import io.druid.query.aggregation.AggregatorFactoryNotMergeableException;
+import io.druid.query.aggregation.AggregatorUtil;
+import io.druid.query.aggregation.BufferAggregator;
+import io.druid.query.cache.CacheKeyBuilder;
+import io.druid.segment.ColumnSelectorFactory;
+import io.druid.segment.ColumnValueSelector;
+import io.druid.segment.column.ValueType;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class DoublesSketchAggregatorFactory extends AggregatorFactory
+{
+
+  private static final int DEFAULT_K = 128;
+
+  // Used for sketch size estimation.
+  private static final long MAX_STREAM_LENGTH = 1_000_000_000;
+
+  private final String name;
+  private final String fieldName;
+  private final int k;
+  private final byte cacheTypeId;
+
+  @JsonCreator
+  public DoublesSketchAggregatorFactory(
+      @JsonProperty("name") final String name,
+      @JsonProperty("fieldName") final String fieldName,
+      @JsonProperty("k") final Integer k)
+  {
+    this(name, fieldName, k, AggregatorUtil.QUANTILES_DOUBLES_SKETCH_BUILD_CACHE_TYPE_ID);
+  }
+
+  DoublesSketchAggregatorFactory(final String name, final String fieldName, final Integer k, final byte cacheTypeId)
+  {
+    if (name == null) {
+      throw new IAE("Must have a valid, non-null aggregator name");
+    }
+    this.name = name;
+    if (fieldName == null) {
+      throw new IAE("Parameter fieldName must be specified");
+    }
+    this.fieldName = fieldName;
+    this.k = k == null ? DEFAULT_K : k;
+    Util.checkIfPowerOf2(this.k, "size");
+    this.cacheTypeId = cacheTypeId;
+  }
+
+  @Override
+  public Aggregator factorize(final ColumnSelectorFactory metricFactory)
+  {
+    if (metricFactory.getColumnCapabilities(fieldName) != null
+        && ValueType.isNumeric(metricFactory.getColumnCapabilities(fieldName).getType())) {
+      final ColumnValueSelector<Double> valueSelector = metricFactory.makeColumnValueSelector(fieldName);
+      if (valueSelector == null) {
```

Comment:
I it seems to me that this can happen if a non-existent field is mentioned as the input. I am not convinced myself that we really need this special no-op aggregator. This is how it was done in the Theta sketch aggregator. I think that it is not worth optimizing an erroneous query.

Reply:
In Druid, ```NilColumnValueSelector``` is returned for non-existent input fields. I think it's worthwhile to optimize, but we already have ```NoopAggregator``` and ```NoopBufferAggregator```, and you can use them instead of adding new ones.

Reply:
I believe I tested this case and got a null for selector. Regarding the dummy aggregators, I believe the idea was to always return a sketch, even an empty one. The custom dummy aggregators do just that.

Reply:
@jihoonson sketches post aggs expect a sketch object, not sure how a sketches post agg will behave if we use `NoopAggregator` and `NoopBufferAggregator`. This check got added considering you can have a sketch field in some segments(say I added a new sketch column from today onwards) but past data does not have that field.

Reply:
If it's nullable, every other aggregatorFactory should consider as well, but they don't. As I said, if a segment doesn't have a specified column, ```NilColumnValueSelector``` is returned instead of null.

Reply:
Yes, we should check for `NilColumnValueSelector` but will continue to return `DoublesSketchNoOpAggregator` as quantile postAggregators expect sketch values.